lib/util/props.js

```diff
@@ -63,10 +65,20 @@ function isRequiredPropType(propTypeExpression) {
   return propTypeExpression.type === 'MemberExpression' && propTypeExpression.property.name === 'isRequired';
 }
 
+/**
+ * Checks if the PropType is a function.
+ * @param {String} propType to check.
+ * @returns {Boolean} `true` if the PropType is a function, `false` if not.
+ */
+function isPropTypeAFunction(propType) {
+  return PROP_TYPE_FUNCTIONS.indexOf(propType) > -1;
```

Comment:
i'm a bit confused here; are we already sure by the time `propType` comes in that it's off of `prop-types`? Or if i have `import { shape } from 'airbnb-prop-types`, will that trigger it?

Reply:
At this point `propType` can be from anywhere - I hadn't considered the scenario of named custom validators like `airbnb-prop-types`. So yes, the example you mentioned would also trigger this error.

I'm not sure how to go about avoiding that 😕 Do you have any ideas?

Reply:
Track where it's imported/required from, and ensure it's `'prop-types'`?

Reply:
I've given that a try 😄